src/net.cpp

```diff
@@ -262,7 +260,7 @@ std::optional<CAddress> GetLocalAddrForPeer(CNode *pnode)
     if (addrLocal.IsRoutable() || gArgs.GetBoolArg("-addrmantest", false))
     {
         LogPrint(BCLog::NET, "Advertising address %s to peer=%d\n", addrLocal.ToString(), pnode->GetId());
-        return addrLocal;
+        return {addrLocal};
```

Comment:
4c59e37c18b177fc50786de37ffffb52de6daaa7: Is this change needed? For me it compiles without.